app/src/main/java/caller/alarmist/AlarmClockWatcher.java

```diff
@@ -373,6 +374,14 @@ public void onCreate() {
         notify.createNotificationChannel(notify_channel);
         notify.createNotificationChannel(service_notify_channel);
 
+        notification = new Notification.Builder(ctx, ALARMIST_SERVICE_CHANNEL)
+                .setContentTitle("Alarmist")
+                .setContentText("Alarmist notification service.")
+                .setTicker("Alarmist")
+                .setSmallIcon(R.drawable.stat_notify_alarm)
+                .build();
+
+        startForeground(NOTIFICATION_SERVICE_ID, notification);
```

Comment:
So is it the case that on Oreo we can't run a NotificationListenerService without a notification?

Reply:
More that we can't run a service that will survive no longer having Settings open without a notification.